### ISSUE TITLE
I would like `rumdl check --fix` to return 0 if no violations remain after code fixes

### ISSUE THREAD

**Issue by hituzi-no-sippo** [rvben/rumdl#197]
Even when no violations remain after code fixes, `rumdl check --fix` returns 1.

If this is not the intended behavior, I would like it to behave the same as markdownlint.
I would like `rumdl check --fix` to return 0 if no violations remain after code fixes.

I read the following, but I couldn't determine if returning 1 when no violations remain is the intended behavior:

- README
  https://github.com/rvben/rumdl/blob/6f0b710b7b7148e34055284e5778b79f0e46331f/README.md?plain=1#L29-L30
- test
  https://github.com/rvben/rumdl/blob/6f0b710b7b7148e34055284e5778b79f0e46331f/tests/cli_integration_tests.rs#L1558-L1611
- #134

## Current Behavior (v0.0.194)

``` console
$ rumdl version
rumdl 0.0.194
$ cat .rumdl.toml
[MD007]
indent = 2
$ cat foo.md
# Heading 1

- list 1
    - list 2
$ rumdl check --fix foo.md
foo.md:4:1: [MD007] Expected 2 spaces for indent depth 1, found 4 [fixed]

Fixed: Fixed 1/1 issues in 1 file (28ms)
$ echo $?
1 # expected `0`
$ rumdl check foo.md

Success: No issues found in 1 file (26ms)
$ echo $?
0
$ cat foo.md
# Heading 1

- list 1
  - list 2
```

When running `rumdl check --fix` on files with no violations, it returns 0.

``` console
$ rumdl check foo.md

Success: No issues found in 1 file (18ms)
$ echo $?
0
$ rumdl check --fix foo.md

Success: No issues found in 1 file (16ms)
$ echo $?
0
```

## markdownlint behavior (for comparison)

markdownlint returns 0 after fixing violations:

``` console
$ markdownlint --version
0.47.0
$ cat foo.md
# Heading 1

- list 1
    - list 2
$ markdownlint foo.md
foo.md:4:1 error MD007/ul-indent Unordered list indentation [Expected: 2; Actual: 4]
$ echo $?
1
$ markdownlint --fix foo.md
$ echo $?
0
$ markdownlint foo.md
$ echo $?
0
$ cat foo.md
# Heading 1

- list 1
  - list 2
```

**Comment by rvben** [comment]
Hi @hituzi-no-sippo,

Thanks for raising this. This indeed should be the same as ruff's and markdownlint's behavior. This will be corrected in upcoming release.

**Comment by martimlobao** [comment]
> Hi [@hituzi-no-sippo](https://github.com/hituzi-no-sippo),
> 
> Thanks for raising this. This indeed should be the same as ruff's and markdownlint's behavior. This will be corrected in upcoming release.

@rvben @hituzi-no-sippo One thing to consider though is that it's not guaranteed that `--fix` results in a state where no additional fixes are needed, even if all the pre-existing issues were addressed. I've had a situation recently where running `--fix` kept resolving existing errors and creating new issues at the same time (iirc, it was a back-and-forth between MD005 and MD007).

I don't think I'd want `--fix` to return 0 if there were new issues created.

**Comment by rvben** [comment]
The issue has been fixed in **version 0.0.195**. 

Thanks for the concern @martimlobao. The logic should prevent returning 0 when there are still issues left. If you hit unexpected results, let me know.

Btw, if fixes oscillate between two states, there's a bug, because  fixes should be idempotent and converge to a stable state. If that happens, please raise an issue.

Thanks again @hituzi-no-sippo for raising this. 🚀

**Comment by martimlobao** [comment]
@rvben I've found a situation where `rumdl fmt` (v0.0.196) does not apply all the fixes in one go. It doesn't go back and forth like I said it would, but it does create _new_ issues that didn't exist on the first run. It eventually does converge though if you run `rumdl fmt` enough times. 

See #209.

I don't know if this is intended behavior or not. I'm personally fine with having to run `fmt` more than once if it eventually converges, though ideally it wouldn't exit in a success state if there are new issues to be fixed.

**Comment by rvben** [comment]
Thanks @martimlobao, I'll investigate and we'll track this in #209.
